### PR TITLE
Packman browse fixes

### PR DESCRIPTION
**Pull request by live627**
- Download some random mods
- Download at least four avatar packs, install any two
- Sort by installed time, nothing happens
- Apply commits one and two. Sorting avatars works, except that some vanish.
- Apply the fifth commit. Sort avatars again. All four avatars now show.
- Download even more mods. Change their type to randomness. You can leave then unzipped.
- Observe the unknown packages all are enumerated as 1. The third commit fixes this.

### CODE REVIEW

File: Sources/Packages.php [hunk 1]
```diff
@@ -1356,6 +1356,10 @@ function PackageBrowse()
 	$context['page_title'] .= ' - ' . $txt['browse_packages'];
 
 	$context['forum_version'] = $forum_version;
+	$context['available_modification'] = array();
+	$context['available_avatar'] = array();
+	$context['available_language'] = array();
+	$context['available_unknown'] = array();
```
Comment (Sesquipedalian) [inline Sources/Packages.php:1365]: After resolving conflicts, I suggest removing these four lines and instead using a foreach to loop through `$context['modification_types']` to set these variables. If you put that loop after the hook, then we'll be able to handle any types added by the hook, too.

Comment (live627) [reply Sources/Packages.php:1365]: Eh,  like this?

```	call_integration_hook('integrate_modification_types');

	require_once($sourcedir . '/Subs-List.php');
	foreach ($context['modification_types'] as $type)
	{
		$context['available_' . $type] = array();

		// Use the standard templates for showing this.
		$listOptions = array(
```

Comment (Sesquipedalian) [reply Sources/Packages.php:1365]: Yes.

**Comment by live627** [comment]
Ugh..

**Comment by Sesquipedalian** [comment]
> Sesquipedalian requested this change. It makes no sense to me.

Er, that might be because the content of 569c515 is not the change I requested. I was asking for this:
```
function PackageBrowse()
{
	global $txt, $scripturl, $context, $sourcedir, $smcFunc;

	$context['page_title'] .= ' - ' . $txt['browse_packages'];
	$context['forum_version'] = SMF_FULL_VERSION;
	$context['modification_types'] = array('modification', 'avatar', 'language', 'unknown');

	call_integration_hook('integrate_modification_types');

	require_once($sourcedir . '/Subs-List.php');

	foreach ($context['modification_types'] as $type)
	{
		$context['available_' . $type] = array();

		// Use the standard templates for showing this.
		...
```

**Comment by live627** [comment]
Ahem.

Your requested change could possibly introduce an edge case issue. Seems that your goal is for hooks to not have to declare `available_trash`. So I sorted that  in the best possible way I knew how . Declare the variable in the same function that writes to it.

My original solution simply declared the four empty context arrays before the hook is called. Which was mostly moving them from after the lists were created, where they were just dead weight.

The hooked functions already have to declare `$global $context` so that `$context['modification_types']`. Was your intention to not have to bring `$context` into scope? That'll end well...

I do not want to revisit this issue in the future. This aims to fix bugs, not create new ones. Ahem.

I'm out. Maybe that wasn't enough time for me to simmer down.

**Comment by Sesquipedalian** [comment]
You make a good point. Please feel free to revert back to your original solution.

**Comment by live627** [comment]
Commit undone.

**Pull request merged**